main/src/ca/uwaterloo/flix/language/ast/UnkindedType.scala

```diff
@@ -0,0 +1,267 @@
+package ca.uwaterloo.flix.language.ast
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+object UnkindedType {
+  case class Var(sym: Symbol.UnkindedTypeVarSym, loc: SourceLocation) extends UnkindedType {
+    override def equals(that: Any): Boolean = that match {
```

Comment:
I chose this way because the default cases scare me :)

```scala
(t1, t2) match {
    case (Var, Var) => ...
    case _ => ...
    // oops forgot Arrow
}
```

Reply:
But that would be an exception right?

Reply:
But I see the point.